sky/serve/replica_managers.py

```diff
@@ -496,11 +517,21 @@ def __init__(self, service_name: str, spec: 'service_spec.SkyServiceSpec',
 
     def get_ready_replica_urls(self) -> List[str]:
         ready_replica_urls = []
-        infos = serve_state.get_replica_infos(self._service_name)
-        for info in infos:
+        version2url = collections.defaultdict(list)
+        for info in serve_state.get_replica_infos(self._service_name):
             if info.status == serve_state.ReplicaStatus.READY:
                 assert info.url is not None
+                version2url[info.version].append(info.url)
                 ready_replica_urls.append(info.url)
+        # Try all version in descending order. There is possibility that
+        # user consecutively update the service several times, and some
+        # version might not have any ready replicas.
+        version = self.latest_version
+        while version >= serve_constants.INITIAL_VERSION:
+            if version in version2url:
+                return version2url[version]
+            version -= 1
+        # If not separate replicas, return all ready replicas.
         return ready_replica_urls
```

Comment:
```suggestion
        return []
```
should we return empty list here?